openlibrary/core/bookshelves.py

```diff
@@ -106,8 +109,21 @@ def most_logged_books(
         where = 'WHERE bookshelf_id' + ('=$shelf_id' if shelf_id else ' IS NOT NULL ')
         if since:
             where += ' AND created >= $since'
-        query = f'select work_id, count(*) as cnt from bookshelves_books {where}'
-        query += ' group by work_id order by cnt desc limit $limit offset $offset'
+        if minimum:
+            minimum = f"HAVING COUNT(*) > {minimum}"
+        groups = [f'work_id {minimum}']
+        group_by = f'group by {", ".join(groups)}'
+        orders = ['cnt desc']
+        if distinct:
+            orders = ['cnt desc'] + orders
+            distinct = 'distinct on (work_id)'
+        order_by = f'order by {", ".join(orders)}'
+        if not sort_by_count:
+            order_by = ''
+        select = (
+            f'select {distinct or ""} work_id, count(*) as cnt from bookshelves_books'
+        )
```

Comment:
Avoid converting a bool to a str.
```suggestion
            distinct_str = 'distinct on (work_id)'
        else:
            distinct_str = ''
        order_by = f'order by {", ".join(orders)}'
        if not sort_by_count:
            order_by = ''
        select = (
            f'select {distinct_str} work_id, count(*) as cnt from bookshelves_books'
        )
```